src/app/panel/auth/SignUp.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { TextInput, Button } from '@tremor/react';
+import { APIService } from '@/services/api';
+
+const SignUp: React.FC = () => {
+  const [username, setUsername] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [confirmPassword, setConfirmPassword] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string | null>('');
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    APIService.signUp(username, email, password);
+  };
+
+  const onUsernameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
+    setUsername(e.target.value);
+    validateForm();
+  }
+
+  const onEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
+    setEmail(e.target.value);
+    validateForm();
+  } 
+
+  const onPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
+    setPassword(e.target.value);
+    validateForm();
+  }
+
+  const onConfirmPasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.target.value = e.target.value.replace(/[^a-zA-Z0-9]/g, '');
+    setConfirmPassword(e.target.value);
+    validateForm();
+  }
+
+  const validateForm = (): void => {
+    if (username.length < 3) {
+      setIsFormValid(false);
+    }
+    else if (email.length < 3) {
+      setIsFormValid(false);
+    }
+    else if (password !== confirmPassword) {
+      setIsFormValid(false);
+    }
+    else if (password.length < 8) {
+      setIsFormValid(false);
```

Comment:
Why not OR all of these conditions together? :pray:

Reply:
I choose to go this way in `SignUp` component in case we wanted to give different warnings to the user, which are also not added yet ^^